### ISSUE TITLE
Use gerund as user task listener event types

### ISSUE THREAD

**Issue by korthout** [camunda/camunda#25541]
We will use the event names of the transitions (i.e. gerunds '-ing') as names of the User Task Listener event types: `creating`, `assigning`, `updating`, `canceling`, and `completing`.

This better reflects the event that the task listener responds to, that the task hasn't finished transitioning yet, and that the task listener can still prevent or correct it.

```[tasklist]
### Tasks
- [ ] https://github.com/camunda/camunda/issues/25543
- [ ] https://github.com/camunda/camunda/issues/25544
- [ ] https://github.com/camunda/camunda/issues/25545
- [ ] https://github.com/camunda/camunda/issues/25546
```

**Comment by korthout** [comment]
We will **NOT** use the event names of the transitions as names of the User Task Listener event types, but instead continue to use the C7 names: `create`, `assignment`, `update`, `cancel`, and `complete`.

Reasoning: Modeler has already released `5.30.0` introducing `assignment` and `complete` listeners. Changing this now would would make Modeler `5.30.0` incompatible with Camunda `8.7.0` and result in additional complexity in Modeler to migrate values produced in `5.30.0`.

**Comment by korthout** [comment]
Reopened, as we've decided to still make this change.
